cinema/views.py

```diff
@@ -43,6 +52,22 @@ def get_serializer_class(self):
 
         return MovieSerializer
 
+    def get_queryset(self):
+        queryset = self.queryset.prefetch_related("genres", "actors")
+        genres = self.request.query_params.get("genres")
```

Comment:
Do you need this filtering for each _action_ or only for "_list_"? Maybe it's better to check the action name and then do filters.